src/components/storage/CockpitStorageIntegration.jsx

```diff
@@ -247,6 +251,52 @@ const CheckStorageDialog = ({
         }
     }, [useConfiguredStorage, checkStep]);
 
+    useEffect(() => {
+        if (checkStep !== "luks") {
+            return;
+        }
+
+        const cockpitDevices = (
+            Object.keys(newMountPoints)
+                    .map(devicePath => ({
+                        devicePath,
+                        deviceName: getDeviceNameByPath(deviceData, devicePath),
+                    }))
+        );
+
+        const devicesToUnlock = (
+            cockpitDevices
+                    .filter(({ devicePath, deviceName }) => {
+                        return (
+                            newMountPoints[devicePath].type === "crypto" &&
+                            deviceData[deviceName].formatData.attrs.v.has_key !== "True"
+                        );
+                    })
+                    .map(({ devicePath, deviceName }) => ({
+                        deviceName,
+                        passphrase: cockpitPassphrases[devicePath],
+                    }))
+        );
+
+        if (devicesToUnlock.some(dev => !dev.passphrase)) {
+            onCritFail()({ message: _("Cockpit storage did not provide the passphrase to unlock encrypted device.") });
+        }
+
+        if (devicesToUnlock.length === 0) {
+            setCheckStep("prepare-partitioning");
+            return;
+        }
+
+        Promise.all(devicesToUnlock.map(unlockDevice))
```

Comment:
`cockpit_mount_points` does not contain non-filesystem things like LVM2 physical volumes, so if you make a encrypted LVM2 physical volume, `cockpit_passprhases` will contain the passphrase for it, but Anaconda will not unlock it since it isn't also mentioned in `cockpit_mount_points`. What about just unlocking everything that is mentioned in `cockpit_passphrases`, without filtering out devices that are not mentioned in `cockpit_mount_points`?

Reply:
This probably wont work if you exist storage mode and enter again, as it will try to unlock unlocked devices. But I see the approach, will adjust.